### PR TITLE
Add docs for Duo Security login

### PR DESCRIPTION
**Pull request by NipuniBhagya**

### CODE REVIEW

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+In this guide, you'll learn how to integrate Duo with your Asgardeo application, enabling Duo as a second-factor authentication provider to enhance your application's security.
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+- Duo Account: A Duo administrator account to configure your Duo application.
+- Asgardeo Account: Access to your Asgardeo console with administrative privileges.
+
+## Configure Duo Security Application
```
Comment (pamodaaw) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:14]: Can't we link to the Duo documentation for this configuring steps.

Comment (NipuniBhagya) [reply en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:14]: There are no guides provided for this specific flow in the Duo security official documentation. So therefore we may have to provide the step by step guide on our end.

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    ![Create duo security app]({{base_path}}/assets/img/guides/mfa/duo/add-duo-security-app.png){: style="display: block; margin: 0; border: 0.3px solid lightgrey;"}
+
+3. On the **Protect an Application page**, search for **Web SDK** and click on **Protect**.
+
+    ![Protect web SDK app]({{base_path}}/assets/img/guides/mfa/duo/protect-web-sdk-app.png){: style="display: block; margin: 0; border: 0.3px solid lightgrey;"}
+
+4. Save the **Client ID**, **Client secret**, and **API hostname**, as you will need these later.
+
+## Add Duo as a Connection in Asgardeo
+
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:28]: before beginning the steps, add an additional sentence.

Something like "Follow the steps below to...."

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+1. In the [Asgardeo Console](https://console.asgardeo.io/), navigate to **Applications**.
+2. Select the application for which you want to enable Duo MFA.
+3. Go to the **Login Flow** section. Username and Password will be added as the default login method.
+4. Click on the **+ icon** to add a new step to the login flow. 
+5. Click on the **+ Add Sign In Option** button in step 2.
+6. Select the Duo connection from the authenticator list and click **Add**. 
+7. Save the changes by clicking the **Update** button.
+
+## Test the Integration
+
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:49]: Add a sentence before the steps as pointed out in the previous comment.

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Add Duo login
+
+[Duo](https://duo.com/) is a multi factor authentication option that offer multiple verification methods such as push notifications, SMS, phone calls, and hardware tokens.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:3]:
```suggestion
[Duo](https://duo.com/) is a multi-factor authentication option that offer multiple verification methods such as push notifications, SMS, phone calls, and hardware tokens.
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+In this guide, you'll learn how to integrate Duo with your Asgardeo application, enabling Duo as a second-factor authentication provider to enhance your application's security.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:5]:
```suggestion
In this guide, you'll learn how to integrate Duo as a second-factor authentication provider for your Asgardeo application.
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+- [Duo Account](https://signup.duo.com/): A Duo administrator account to configure your Duo application.
+- [Asgardeo Account](https://asgardeo.io/signup): Access to your Asgardeo console with administrative privileges.
+
+## Configure Duo Security Application
+Follow the steps below to configure the Duo Security application.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:15]:
```suggestion

```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+3. On the **Protect an Application page**, search for **Web SDK** and click on **Protect**.
+
+    ![Protect web SDK app]({{base_path}}/assets/img/guides/mfa/duo/protect-web-sdk-app.png){: style="display: block; margin: 0; border: 0.3px solid lightgrey;"}
+
+4. Save the **Client ID**, **Client secret**, and **API hostname**  as you will need these later.
+
+## Add Duo as a Connection in Asgardeo
+
+Follow the steps below to integrate Duo as a connection in Asgardeo.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:31]:
```suggestion
Next, you need to register Duo as a connection in Asgardeo. To do so, follow the steps below.
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Log in to the [Asgardeo Console](https://console.asgardeo.io/) as an adminstrator.
+2. Navigate to **Connections** from the left-hand panel 
+3. Click **New Connection** and select **Duo** from the listed templates.
+4. Provide a Name for the Duo connection (e.g., "Duo Security").
+5. Enter the **Client ID**, **Client secret**, and **API hostname**, that you obtained from the Duo Admin Panel.
+6. Click **Finish** to save the configuration.
+
+## Configure Duo for MFA
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:40]:
```suggestion
## Enable Duo MFA for an app
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+Follow the steps below to add Duo as an MFA option to the login flow of the application.
+
+1. In the [Asgardeo Console](https://console.asgardeo.io/), navigate to **Applications**.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:44]:
```suggestion
1. On the [Asgardeo Console](https://console.asgardeo.io/), navigate to **Applications**.
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+1. Log in to the [Asgardeo Console](https://console.asgardeo.io/) as an adminstrator.
+2. Navigate to **Connections** from the left-hand panel 
+3. Click **New Connection** and select **Duo** from the listed templates.
+4. Provide a Name for the Duo connection (e.g., "Duo Security").
+5. Enter the **Client ID**, **Client secret**, and **API hostname**, that you obtained from the Duo Admin Panel.
+6. Click **Finish** to save the configuration.
+
+## Configure Duo for MFA
+
+Follow the steps below to add Duo as an MFA option to the login flow of the application.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:42]:
```suggestion
The created Duo connection should be added as  an MFA option to the login flow of an application. To do so, follow the steps below.
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+1. Log in to the [Asgardeo Console](https://console.asgardeo.io/) as an adminstrator.
+2. Navigate to **Connections** from the left-hand panel 
+3. Click **New Connection** and select **Duo** from the listed templates.
+4. Provide a Name for the Duo connection (e.g., "Duo Security").
+5. Enter the **Client ID**, **Client secret**, and **API hostname**, that you obtained from the Duo Admin Panel.
+6. Click **Finish** to save the configuration.
+
+## Enable Duo MFA for an application
+
+The created Duo connection should be added as  an MFA option to the login flow of an application. To do so, follow the steps below.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:40]:
```suggestion
The created Duo connection should be added as an MFA option to the login flow of an application. To do so, follow the steps below.
```

File: en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+2. Select the application for which you want to enable Duo MFA.
+3. Go to the **Login Flow** section. Username and Password will be added as the default login method.
+4. Click on the **+ icon** to add a new step to the login flow. 
+5. Click on the **+ Add Sign In Option** button in step 2.
+6. Select the Duo connection from the authenticator list and click **Add**. 
+7. Save the changes by clicking the **Update** button.
+
+## Test the Integration
+
+Finally, you need to verify that Duo has been successfully integrated and is functioning correctly as a second-factor authentication method.
```
Comment (himeshsiriwardana) [inline en/asgardeo/docs/guides/authentication/mfa/add-duo-login.md:52]:
```suggestion
Finally, you need to verify that Duo has been successfully integrated and is functioning correctly as a second-factor authentication method. To do so,
```

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**